orttraining/orttraining/python/training/onnxblock/graph.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# graph.py
+
+from abc import ABC, abstractmethod
+import onnx
+from ._graph_utils import (
+    build_gradient_model,
+    build_gradient_accumulation_model,
+    get_model_parameters,
+)
+
+
+class Graph(ABC):
+    """Builds the forward model based on user's build method."""
+
+    def __init__(self):
+        ...
+
+    @abstractmethod
+    def build(self, *args, **kwargs):
+        """Customize and build the forward graph for this model.
+
+        This method is to be overriden by the user's implementation.
+        """
+        ...
+
+    def __call__(self, *args, **kwargs):
+        """Calls the user's build method and runs validation on top.
+
+        The output onnx model is got by invoking the user's build method.
+        """
+        # build the user model
+        user_model = self.build(*args, **kwargs)
```

Comment:
We have a case where the user does not want to add loss to the graph, but just do the other things- like separate params(eg for constructing an eval graph). In that case should we allow the user to not have to implement the build method- i.e not have an abstract build method here